File: Fundamentals/ActivityLifecycle/app/src/main/java/com/outlook/gonzasosa/activitylifecycle/MainActivity.java
package com.outlook.gonzasosa.activitylifecycle;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private final String TAG = "PKAT";

    @Override
    protected void onCreate (Bundle savedInstanceState) {
        super.onCreate (savedInstanceState);
        Log.d (TAG, "onCreate");
    }

    @Override
    protected void onStart () {
        super.onStart ();
        Log.d (TAG, "onStart");
    }

    @Override
    protected void onResume () {
        super.onResume ();
        Log.d (TAG, "onResume");
    }

    @Override
    protected void onPause () {
        super.onPause ();
        Log.d (TAG, "onPause");
    }

    @Override
    protected void onStop () {
        super.onStop ();
        Log.d (TAG, "onStop");
    }

    @Override
    protected void onRestart () {
        super.onRestart ();
        Log.d (TAG, "onRestart");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: Fundamentals/ActivityLifecycleKts/app/src/main/java/com/outlook/gonzasosa/activitylifecyclekts/MainActivity.kt
package com.outlook.gonzasosa.activitylifecyclekts

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    val tag = "PKAT"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(tag, "OnCreate")
    }

    override fun onStart() {
        super.onStart()
        Log.d(tag, "onStart")
        
    }

    override fun onResume() {
        super.onResume()
        Log.d(tag, "OnResume")
    }

    override fun onPause() {
        super.onPause()
        Log.d(tag, "OnPause")
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(tag, "OnRestart")
    }

    override fun onStop() {
        super.onStop()
        Log.d(tag, "OnStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d(tag, "OnDestroy")
    }

}